securitas/tests/unit/controller/test_password.py

```diff
@@ -9,3 +11,117 @@ def test_password_reset(client):
     page = BeautifulSoup(result.data, 'html.parser')
     assert page.title
     assert page.title.string == 'Password Reset - The Fedora Project'
+
+
+def test_missing_credentials(client):
+    """Verify that missing details are caught"""
+    result = client.post(
+        '/password-reset',
+        data={
+            "username": "",
+            "current_password": "",
+            "password": "",
+            "password_confirm": "",
+        },
+        follow_redirects=True,
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    messages = page.select(".flash-messages .red")
+    assert len(messages) == 1
+    assert messages[0].get_text(strip=True) == 'Please fill in all fields to reset your password'
+
+
+def test_non_matching_passwords(client):
+    """Verify that passwords that dont match are caught"""
+    result = client.post(
+        '/password-reset',
+        data={
+            "username": "jbloggs",
+            "current_password": "sdsds",
+            "password": "password1",
+            "password_confirm": "password2",
+        },
+        follow_redirects=True,
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    messages = page.select(".flash-messages .red")
+    assert len(messages) == 1
+    assert messages[0].get_text(strip=True) == 'Password and confirmation did not match.'
+
+
+def test_password_policy(client):
+    """Verify that password policies are upheld"""
+    result = client.post(
+        '/password-reset',
+        data={ 
+            "username": "jbloggs",
+            "current_password": "1",
+            "password": "LongSuperSafePassword",
+            "password_confirm": "LongSuperSafePassword",
+        },
+        follow_redirects=True,
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    messages = page.select(".flash-messages .red")
+    assert len(messages) == 1
+    assert messages[0].get_text(strip=True) == 'Failed to reset your password (invalid current password).'
+
+@pytest.mark.vcr()
+def test_time_sensitive_password_policy(client, dummy_user):
+    """Verify that new password policies are upheld"""
+    result = client.post(
+        '/password-reset',
+        data={
+            "username": "dummy",
+            "current_password": "dummy_password",
+            "password": "1",
+            "password_confirm": "1",
+        },
+        follow_redirects=True,
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    messages = page.select(".flash-messages .red")
+    assert len(messages) == 1
+    assert messages[0].get_text(strip=True) == 'Failed to reset your password (policy error): Constraint violation: Too soon to change password'
```

Comment:
According to [this IPA doc](https://www.freeipa.org/page/V2/Group_Password_Policy) the setting that forces an interval of time during two password changes is `krbMinPwdLife` and can be changed via a command like `ipa pwpolicy-mod --minlife=0`. I suggest that for the tests you add a fixture that would set this value to zero and restore it afterwards (maybe with a `session` scope so it's only done once for the whole test suite).

To do that you could just run the command but you'll have to login using kerberos first and that might become complex, and on top of that it won't work on CI where FreeIPA will not be installed. That's why I suggest you use the `ipa_admin` client to set this value and unset it. Unfortunately, the client in `python_freeipa` does not seem to support this command, so the only solution I can think of right now would be to add the command to the subclass I've already written in `security/ipa.py` to add a missing command I needed earlier.